templates/common/aws/units/afterburn-hostname.service.yaml

```diff
@@ -0,0 +1,17 @@
+name: afterburn-hostname.service
+enabled: true
+contents: |
+  [Unit]
+  Description=Afterburn Hostname for AWS
+  After=NetworkManager-wait-online.service
+  Before=node-valid-hostname.service
+
+  [Service]
+  Type=oneshot
+  RemainAfterExit=yes
+  ExecStartPre=/usr/bin/afterburn --provider aws --hostname=/run/afterburn.hostname
+  # Manually set current hostname at /proc/sys/kernel/hostname to correct value for kubelet.service
+  ExecStart=/bin/bash -c "cp /run/afterburn.hostname /proc/sys/kernel/hostname; cp /run/afterburn.hostname /etc/hostname"
```

Comment:
`/proc/sys/kernel/hostname` is the default and only place where golang `os.Hostname` used in kubelet will gain the value. `node-valid-hostname.service` `hostnamectl` call fails to update current hostname in running machine, so there is a manual copy.

```
Could not set property: Connection timed out
```

Reply:
@rphillips Here is an issue with hostnamectl I observed, re: https://github.com/openshift/machine-config-operator/pull/2401#issuecomment-789739611